File: ltexls/src/test/java/org/bsplines/ltexls/server/DelayedDiagnosticsPublisherRunnableTest.java
/* Copyright (C) 2020 Julian Valentin, LTeX Development Community
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package org.bsplines.ltexls.server;

import java.util.concurrent.ExecutionException;
import org.bsplines.ltexls.client.MockLtexLanguageClient;
import org.eclipse.lsp4j.PublishDiagnosticsParams;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class DelayedDiagnosticsPublisherRunnableTest {
  @Test
  public void testRun() throws InterruptedException, ExecutionException {
    LtexLanguageServer languageServer = new LtexLanguageServer();
    MockLtexLanguageClient languageClient = new MockLtexLanguageClient();
    languageServer.connect(languageClient);

    LtexTextDocumentItem document = new LtexTextDocumentItem(
        languageServer,"untitled:text.md", "markdown", 1, "This is an test.\n");
    document.checkAndPublishDiagnosticsWithCache();

    DelayedDiagnosticsPublisherRunnable runnable = new DelayedDiagnosticsPublisherRunnable(
        languageClient, document);
    runnable.run();

    Assertions.assertEquals(2, languageClient.getPublishDiagnosticsParamsList().size());

    for (PublishDiagnosticsParams publishDiagnosticsParams :
          languageClient.getPublishDiagnosticsParamsList()) {
      Assertions.assertEquals("untitled:text.md", publishDiagnosticsParams.getUri());
      Assertions.assertEquals(1, publishDiagnosticsParams.getDiagnostics().size());
    }
  }
}

File: src/test/kotlin/org/bsplines/ltexls/server/DelayedDiagnosticsPublisherRunnableTest.kt
/* Copyright (C) 2019-2023 Julian Valentin, LTeX Development Community
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package org.bsplines.ltexls.server

import org.bsplines.ltexls.client.MockLtexLanguageClient
import org.eclipse.lsp4j.PublishDiagnosticsParams
import kotlin.test.Test
import kotlin.test.assertEquals

class DelayedDiagnosticsPublisherRunnableTest {
  @Test
  fun testRun() {
    val languageServer = LtexLanguageServer()
    val languageClient = MockLtexLanguageClient()
    languageServer.connect(languageClient)

    val document = LtexTextDocumentItem(
      languageServer,
      "untitled:test.md",
      "markdown",
      1,
      "This is an test.\n",
    )
    document.checkAndPublishDiagnosticsWithCache()

    val runnable = DelayedDiagnosticsPublisherRunnable(languageClient, document)
    runnable.run()

    assertEquals(2, languageClient.publishDiagnosticsParamsList.size)

    for (
      publishDiagnosticsParams: PublishDiagnosticsParams in
      languageClient.publishDiagnosticsParamsList
    ) {
      assertEquals("untitled:test.md", publishDiagnosticsParams.uri)
      assertEquals(1, publishDiagnosticsParams.diagnostics.size)
    }
  }
}
